presto-main/src/main/java/com/facebook/presto/metadata/FunctionSignatureResolver.java

```diff
@@ -0,0 +1,325 @@
+/*
```

Comment:
Is it just a refactor? Or did you have more changes too? If it's just a simple refactor, may be move it out.